service/remote.go

```diff
@@ -140,24 +141,46 @@ func (r *RemoteService) AddRemoteBindingListener(bindingListener cluster.RemoteB
 func (r *RemoteService) Call(ctx context.Context, req *protos.Request) (*protos.Response, error) {
 	c, err := util.GetContextFromRequest(req, r.server.ID)
 	c = util.StartSpanFromRequest(c, r.server.ID, req.GetMsg().GetRoute())
+	defer tracing.FinishSpan(c, err)
 	var res *protos.Response
+
+	if err == nil {
+		result := make(chan *protos.Response, 1)
+		go func() {
+			result <- processRemoteMessage(c, req, r)
+		}()
+
+		// Set a timeout for processing the call
+		timeout := time.Duration(300) * time.Second
+		if ctx.Value("reqTimeout") != nil {
+			timeout, err = time.ParseDuration(fmt.Sprint(ctx.Value("reqTimeout")))
+			if err != nil {
+				logger.Log.Errorf("Error while parsing timeout duration: %s", err.Error())
+			}
+		}
+
+		select {
+		case <-time.After(timeout):
+			err = fmt.Errorf("Timed out calling route %s after %d seconds.", req.GetMsg().GetRoute(), timeout)
```

Comment:
In this case, will the request be cancelled or will it keep hanging until something happens? I was expecting to see a context.Cancell or something else

Reply:
Currently, the function called by the route will keep running, but the processing slot will be freed up preventing the queue to fill. Ideally you're right, it should cancel the context as well, just to ensure resources are freed up immediately. I'll add the required changes :)